src/server/pkg/errutil/errutil.go

```diff
@@ -30,3 +30,11 @@ func IsNotFoundError(err error) bool {
 	}
 	return strings.Contains(err.Error(), "not found")
 }
+
+// IsInvalidNameError returns true if err is due to an invalid name
+func IsInvalidNameError(err error) bool {
+	if err == nil {
+		return false
+	}
+	return strings.Contains(err.Error(), "only alphanumeric characters, underscores, and dashes are allowed")
+}
```

Comment:
Could we move this predicate to `src/server/pkg/ancestry/ancestry.go` for same reason we relocated the other one. It seems like there's a particular risk of this one changing since it enumerates the valid characters.